tests/network/l2_bridge/libl2bridge.py

```
@@ -236,18 +258,82 @@ def hot_plug_interface_and_set_address(
+
+    Returns:
+        A ResourceField with interface data gathered from the guest, or None.
+    """
+    # VMI spec interfaces are set by virt-controller seconds after the hot-plug request;
+    # by the time we get here, lookup_iface_status has already waited 2 min for guest-agent.
+    vmi_iface = _lookup_vmi_interface(vmi=vm.vmi, interface_name=interface_name)
+    if not vmi_iface:
+        return None
+    LOGGER.warning(
+        f"CNV-77961: Guest agent did not report interface {interface_name} on VM {vm.name}, "
+        f"falling back to console lookup by MAC {vmi_iface['macAddress']}."
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)